consumer/statistics/reporter_test.go

```diff
@@ -56,70 +53,67 @@ func mockLocationDetector() location.Location {
+	mockSender := newMockRemoteSender()
+	reporter := NewSessionStatisticsReporter(statisticsTracker, mockSender, mockSignerFactory, mockLocationDetector, time.Minute)
 
 	reporter.ConsumeStateEvent(mockStateEvent)
 
 	reporter.start(mockStateEvent.SessionInfo.ConsumerID, mockStateEvent.SessionInfo.Proposal.ServiceType, mockStateEvent.SessionInfo.Proposal.ProviderID, mockStateEvent.SessionInfo.SessionID)
-	assert.NoError(t, waitFor(func() bool { return atomic.LoadInt64(&counter) == 0 }))
-	assert.True(t, reporter.started)
 	reporter.stop()
 
-	assert.NoError(t, waitFor(func() bool { return atomic.LoadInt64(&counter) == 1 }))
+	assert.NoError(t, waitForChannel(mockSender.called, time.Millisecond*200))
 	assert.False(t, reporter.started)
 }
 
 func TestStatisticsReporterInterval(t *testing.T) {
-	var counter int64
-	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		atomic.AddInt64(&counter, 1)
-	}))
-	defer ts.Close()
-
-	mysteriumClient := server.NewClient(ts.URL)
+	mockSender := newMockRemoteSender()
 	statisticsTracker := NewSessionStatisticsTracker(time.Now)
-	reporter := NewSessionStatisticsReporter(statisticsTracker, mysteriumClient, mockSignerFactory, mockLocationDetector, time.Nanosecond)
+	reporter := NewSessionStatisticsReporter(statisticsTracker, mockSender, mockSignerFactory, mockLocationDetector, time.Nanosecond)
 
 	reporter.ConsumeStateEvent(mockStateEvent)
 
 	reporter.start(mockStateEvent.SessionInfo.ConsumerID, mockStateEvent.SessionInfo.Proposal.ServiceType, mockStateEvent.SessionInfo.Proposal.ProviderID, mockStateEvent.SessionInfo.SessionID)
-	assert.NoError(t, waitFor(func() bool { return atomic.LoadInt64(&counter) > 3 }))
+	assert.NoError(t, waitForChannel(mockSender.called, time.Millisecond*200))
 
 	reporter.stop()
 }
 
 func TestStatisticsReporterConsumeStateEvent(t *testing.T) {
-	var counter int64
-	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		atomic.AddInt64(&counter, 1)
-	}))
-	defer ts.Close()
-
-	mysteriumClient := server.NewClient(ts.URL)
+	mockSender := newMockRemoteSender()
 	statisticsTracker := NewSessionStatisticsTracker(time.Now)
-	reporter := NewSessionStatisticsReporter(statisticsTracker, mysteriumClient, mockSignerFactory, mockLocationDetector, time.Nanosecond)
+	reporter := NewSessionStatisticsReporter(statisticsTracker, mockSender, mockSignerFactory, mockLocationDetector, time.Nanosecond)
 	reporter.ConsumeStateEvent(mockStateEvent)
+	<-mockSender.called
 	assert.True(t, reporter.started)
 	copy := mockStateEvent
 	copy.State = connection.Disconnecting
 	reporter.ConsumeStateEvent(copy)
 	assert.False(t, reporter.started)
 }
 
-func waitFor(f func() bool) error {
-	timeout := time.Now().Add(time.Second)
-	for time.Now().Before(timeout) {
-		if f() {
-			return nil
-		}
+func waitForChannel(ch chan bool, duration time.Duration) error {
```

Comment:
We could have this as a general function somewhere in utils.

Reply:
Probably. But not worth it at the moment.